### PR TITLE
Attempt to fix flaky tests regarding Text file busy error

### PR DESCRIPTION
**Pull request by allada**
This change is [Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274)

### CODE REVIEW

**Comment by MarcusSorealheis** [comment]
I read something interesting today about Unix filesystems that is dated and from the Golang repo:

https://github.com/golang/go/issues/22315

And another related issue:

https://github.com/golang/go/issues/22220

File: cas/worker/tests/running_actions_manager_test.rs [hunk 1]
```diff
@@ -1106,14 +1106,19 @@ exit 0
                 .await?
                 .write_all(TEST_WRAPPER_SCRIPT_CONTENT.as_bytes())
                 .await?;
+            #[cfg(target_family = "unix")]
+            test_wrapper_script_handle
+                .as_writer()
+                .await?
+                .as_mut()
+                .set_permissions(Permissions::from_mode(0o755))
+                .await?;
```
Comment (MarcusSorealheis) [inline cas/worker/tests/running_actions_manager_test.rs:1121]: Is there a reason we do not need to include error handling here for the futures like we did in other places with:

```rs 
.await?
.err_tip(|| "Could not sync file");
```

**Review by MarcusSorealheis — APPROVED** [review]
In either case, LGTM.

**Review by chrisstaite-menlo — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274)* status: all files reviewed, 1 unresolved discussion (waiting on @allada)

**Review by allada — COMMENTED** [review]
I believe I found the problem, it's `.flush()`. If you write to a file, and then close it, it will flush, but it does it in a background thread in tokio. To get around this, you need to flush then let the Drop happen.
see: https://github.com/tokio-rs/tokio/issues/2307

> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274#-:-Ne9D0QTBXNUQg9NTD8R:bha9s9h)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @chrisstaite-menlo and @MarcusSorealheis)

**Comment by MarcusSorealheis** [comment]
> I believe I found the problem, it's `.flush()`. If you write to a file, and then close it, it will flush, but it does it in a background thread in tokio. To get around this, you need to flush then let the Drop happen. see: [tokio-rs/tokio#2307](https://github.com/tokio-rs/tokio/issues/2307)
> 
> > _[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274#-:-Ne9D0QTBXNUQg9NTD8R:bha9s9h)_ status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @chrisstaite-menlo and @MarcusSorealheis)

Interesting. Nice find. I was playing with `.flush()` in my investigation.

**Review by chrisstaite-menlo — APPROVED** [review]
Excellent find.

> Reviewed 5 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274#-:-Ne9JnUBA24ElcLrU-n4:b-3n7a8q)* status: all files reviewed, 1 unresolved discussion (waiting on @MarcusSorealheis)

**Review by allada — COMMENTED** [review]
> Dismissed @MarcusSorealheis from a discussion.
> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @allada)

**Review by MarcusSorealheis — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274#-:-Ne9XQWj3BiXzCR8EzgO:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @allada)

**Review by MarcusSorealheis — APPROVED** [review]
LGTM
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274#-:-Ne9XiW43ej9eXIIy9o_:bb6ywmx)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @allada)

**Review by allada — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274)* status: 5 of 6 files reviewed, all discussions resolved (waiting on @chrisstaite-menlo)

**Review by allada — COMMENTED** [review]
> Reviewed 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/allada/turbo-cache/274)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @allada)

**Pull request merged**